src/response.rs

```diff
@@ -212,21 +202,7 @@ impl Response {
     /// [`serde_json::from_reader`]: https://docs.serde.rs/serde_json/fn.from_reader.html
     #[inline]
     pub fn json<T: DeserializeOwned>(&mut self) -> ::Result<T> {
-        // There's 2 ways we could implement this:
-        //
-        // 1. Just using from_reader(self), making use of our blocking read adapter
-        // 2. Just use self.inner.json().wait()
-        //
-        // Doing 1 is pretty easy, but it means we have the `serde_json` code
-        // in more than one place, doing basically the same thing.
-        //
-        // Doing 2 would mean `serde_json` is only in one place, but we'd
-        // need to update the sync Response to lazily make a blocking read
-        // adapter, so that our `inner` could possibly still have the original
-        // body.
-        //
-        // Went for easier for now, just to get it working.
-        serde_json::from_reader(self).map_err(::error::from)
+        self.inner.json().wait()
```

Comment:
The comment here was old, instead of calling `.wait`, it'd be better to use the helper from the `wait` module.

Reply:
Changed to use `wait::timeout()`